test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/DefaultHtmlGeneratorTest.cs

```diff
@@ -190,6 +191,103 @@ public void GenerateTextArea_WithNullExpression_WithNameAttribute_DoesNotThrow()
             Assert.Equal(expected, attribute.Value);
         }
 
+        [Theory]
+        [MemberData(nameof(GetMaxLengthAttributeRenderingTestsData))]
+        public void Generate_RendersMaxLengthAttribute_AsRequested(
+            string expression,
+            int expectedValue,
```

Comment:
Expectations are usually at the end of the theory data. Make this last and `maxLengthShouldBeRendered` second-last.